_pydevd_sys_monitoring/_pydevd_sys_monitoring.py

```diff
@@ -172,23 +174,19 @@ def _get_bootstrap_frame(depth: int) -> Tuple[Optional[FrameType], bool]:
 
 # fmt: off
 # IFDEF CYTHON
-# cdef _get_unhandled_exception_frame(int depth):
+# cdef _get_unhandled_exception_frame(exc, int depth):
 # ELSE
-def _get_unhandled_exception_frame(depth: int) -> Optional[FrameType]:
+def _get_unhandled_exception_frame(exc, depth: int) -> Optional[FrameType]:
 # ENDIF
 # fmt: on
     try:
-        result = _thread_local_info.f_unhandled
-
-        # Make sure the result is from the same exception. That means the result is in the stack somewhere.
-        if result is not None:
-            orig = frame = _getframe(depth)
-            while result != frame and frame is not None:
-                frame = frame.f_back
-            if frame is not None:
-                return result
-            del _thread_local_info.f_unhandled
-            raise AttributeError("Unhandled frame from different exception")
+        # Unhandled frame has to be from the same exception.
+        if _thread_local_info.f_unhandled_exc is exc:
```

Comment:
Debugpy tests were failing when I ported the pydevd changes over. It was because the cached frame was from a previous exception. I added this to make sure the cached item was the same exception. This makes the debugpy tests pass. I updated the tests for this in pydevd to cause this too.